pkg/apply/solver/solver.go

```diff
@@ -211,34 +211,30 @@ func (t *TaskQueueBuilder) Build(taskContext *taskrunner.TaskContext, o Options)
 		}
 	}
 
-	// TODO: add InvSetTask when Destroy=true to retain undeleted objects
-	if !o.Destroy {
-		klog.V(2).Infoln("adding inventory set task")
-		prevInvIds, _ := t.InvClient.GetClusterObjs(t.invInfo)
-		tasks = append(tasks, &task.InvSetTask{
-			TaskName:      "inventory-set-0",
-			InvClient:     t.InvClient,
-			InvInfo:       t.invInfo,
-			PrevInventory: prevInvIds,
-			DryRun:        o.DryRunStrategy,
-		})
+	prevInvIds, _ := t.InvClient.GetClusterObjs(t.invInfo)
+	klog.V(2).Infoln("adding delete/update inventory task")
+	var taskName string
+	if o.Destroy {
+		taskName = "inventory-delete-or-update-0"
 	} else {
-		klog.V(2).Infoln("adding delete inventory task")
-		tasks = append(tasks, &task.DeleteInvTask{
-			TaskName:  "delete-inventory-0",
-			InvClient: t.InvClient,
-			InvInfo:   t.invInfo,
-			DryRun:    o.DryRunStrategy,
-		})
+		taskName = "inventory-set-0"
 	}
+	tasks = append(tasks, &task.DeleteOrUpdateInvTask{
+		TaskName:      taskName,
+		InvClient:     t.InvClient,
+		InvInfo:       t.invInfo,
+		PrevInventory: prevInvIds,
+		DryRun:        o.DryRunStrategy,
+		Destroy:       o.Destroy,
+	})
 
 	return &TaskQueue{tasks: tasks}
 }
 
 // AppendApplyTask appends a task to the task queue to apply the passed objects
 // to the cluster. Returns a pointer to the Builder to chain function calls.
 func (t *TaskQueueBuilder) newApplyTask(applyObjs object.UnstructuredSet,
-	applyFilters []filter.ValidationFilter, applyMutators []mutator.Interface, o Options) taskrunner.Task {
+		applyFilters []filter.ValidationFilter, applyMutators []mutator.Interface, o Options) taskrunner.Task {
```

Comment:
is this a format on save thing? I don't know if a go formatting standard was chosen for this project, but this looks inconsistent.

Reply:
maybe a go version change in fmt?

Reply:
yeah probably my IDE being opinionated

Reply:
reverted